diff --git a/paml/lib/sample_arrays.py b/paml/lib/sample_arrays.py
index 6828549..a2e2ae4 100644
--- a/paml/lib/sample_arrays.py
+++ b/paml/lib/sample_arrays.py
@@ -62,6 +62,7 @@ p = paml.Primitive('ContainerSet')
 p.description = 'Create a new sample collection containing a set of replicate slots for every sample in the input'
 p.add_input('quantity', 'http://bioprotocols.org/uml#ValueSpecification')
 p.add_input('specification', 'http://bioprotocols.org/paml#ContainerSpec')
+p.add_input('replicates', 'http://bioprotocols.org/uml#ValueSpecification', optional=True)
 p.add_output('samples', 'http://bioprotocols.org/paml#SampleArray')
 doc.add(p)
 
